### PR TITLE
 Support sxs with old outputs with sync resources only. 

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to https://github.com/pulumi/pulumi/pull/3663 after we discoverd a sxs issue there.

### CODE REVIEW

File: sdk/nodejs/Makefile [hunk 1]
```diff
@@ -42,7 +42,7 @@ install_plugin::
 install:: install_package install_plugin
 
 istanbul_tests::
-	./node_modules/.bin/istanbul test --print none _mocha -- --timeout 15000 'bin/tests/**/*.spec.js'
+	./node_modules/.bin/istanbul test --print none _mocha -- --timeout 15000 'bin/tests/**/unwrap.spec.js'
```
Comment (CyrusNajmabadi) [inline sdk/nodejs/Makefile]:
```suggestion
	./node_modules/.bin/istanbul test --print none _mocha -- --timeout 15000 'bin/tests/**/*.spec.js'
```

File: sdk/nodejs/tests/unwrap.spec.ts [hunk 2]
```diff
@@ -152,52 +162,62 @@ describe("unwrap", () => {
...
             3,
+            [r1, r2],
             [r1, r2]));
 
         describe("with unknowns", () => {
             it("across 'all' without unknowns", testResources(
                 all([Promise.resolve({ a: createOutput(unknown, r1, r2)}), Promise.resolve({ b: createOutput(unknown, r3, r4)})]),
                 undefined,
+                [],
                 [r1, r2, r3, r4]));
```
Comment (CyrusNajmabadi) [inline sdk/nodejs/tests/unwrap.spec.ts]: these tests now validate both behaviors. the old behavior about what resources you coudl get synchronously from an output and what resources you now get asynchronously.

File: sdk/nodejs/output.ts [hunk 3]
```diff
@@ -381,21 +415,31 @@ export function output<T>(val: Input<T | undefined>): Output<Unwrap<T | undefine
...
+        // accommodate outputs from downlevel SxS SDKs.  This ensures that within this package it is
+        // safe to assume the implementation of any Output returned by the `output` function.
+        //
+        // This includes:
+        // 1. that first-class unknowns are properly represented in the system: if this was a
+        //    downlevel output where val.isKnown resolves to false, this guarantees that the
+        //    returned output's promise resolves to unknown.
+        // 2. That the `isSecret` property is available.
+        // 3. That the `.allResources` is available.
+        const allResources = val.allResources ? val.allResources() : Promise.resolve(new Set<Resource>());
```
Comment (lukehoban) [inline sdk/nodejs/output.ts]: Is there a risk that these `val.allResources` tests trigger proxy logic which returns a `Output<undefined>` which is truthy?

Comment (CyrusNajmabadi) [reply sdk/nodejs/output.ts]: yes.  that's a great catch.

**Pull request merged**